grails-app/services/org/pih/warehouse/fulfillment/FulfillmentService.groovy

```diff
@@ -70,4 +101,252 @@ class FulfillmentService {
 
         return command
     }
+
+    StockMovement createOutbound(ImportPackingListCommand command) {
+        // First create a requisition
+        Requisition requisition = createRequisition(command.fulfillmentDetails)
+        // Having a requisition we can create a shipment
+        Shipment shipment = createShipment(command.sendingOptions, requisition)
+        // Create requisition items and group them with the corresponding packing list item, to be able to indicate when creating a picklist item, what bin location
+        // corresponds to a particular requisition item
+        Map<RequisitionItem, ImportPackingListItem> requisitionItemsGrouped = createRequisitionItems(command.packingList, requisition)
+        // Having requisition items, we can build a picklist
+        createPicklist(requisition, requisitionItemsGrouped)
+        // Build shipment items looking at picklist items associated with requisition items
+        createShipmentItems(shipment, requisition)
+        // Create a shipped event, associate it with the shipment, and issue the requisition
+        sendShipment(shipment)
+        // In the end create a debit transaction with its entries
+        createOutboundTransaction(shipment)
+
+        return StockMovement.createFromRequisition(requisition)
+    }
+
+    String generateName(Location origin, Location destination, Date dateRequested, String trackingNumber, String description) {
+        final String separator =
+                Holders.getConfig().getProperty("openboxes.generateName.separator") ?: Constants.DEFAULT_NAME_SEPARATOR
+
+        String originIdentifier = origin?.locationNumber ?: origin?.name
+        String destinationIdentifier = destination?.locationNumber ?: destination?.name
+        StringBuilder name = new StringBuilder("${originIdentifier}${separator}${destinationIdentifier}")
+        if (dateRequested) {
+            name.append("${separator}${dateRequested?.format(Constants.GENERATE_NAME_DATE_FORMAT)}")
+        }
+        if (trackingNumber) {
+            name.append("${separator}${trackingNumber}")
+        }
+        if (description) {
+            name.append("${separator}${description}")
+        }
+        return name
+                .toString()
+                .replace(Constants.SPACE_SEPARATOR, "")
```

Comment:
Not really important but `SPACE_SEPARATOR` is somewhat misleading here since you're actually just trimming all white space from a string and it has nothing to do with the separator character.

I'd make this a `StringUtils.removeWhiteSpace(String s)` utility method that we can re-use anywhere.